File: app/src/test/java/br/com/samueltobias/customercrud/model/CustomerTest.java
package br.com.samueltobias.customercrud.model;

import org.junit.Assert;
import org.junit.Test;

import br.com.samueltobias.customercrud.model.Customer;

public class CustomerTest {

    @Test
    public void shouldReturnValidCustomer_WhenFillingAllFields() {
        Customer customer = new Customer("Mercado XPTO", "99999-9999");

        Assert.assertTrue(customer.isValid());
    }

    @Test
    public void shouldReturnInvalidCustomer_WhenAllFieldsAreEmpty() {
        Customer customer = new Customer("", "");

        Assert.assertFalse(customer.isValid());
    }
}

File: app/src/test/java/br/com/samueltobias/customercrud/domain/model/CustomerTest.kt
package br.com.samueltobias.customercrud.domain.model

import org.junit.Assert
import org.junit.Test

class CustomerTest {
    @Test
    fun shouldReturnValidCustomer_WhenFillingAllFields() {
        val customer = Customer(1, "Mercado XPTO", "99999-9999")
        Assert.assertTrue(customer.isValid)
    }

    @Test
    fun shouldReturnInvalidCustomer_WhenAllFieldsAreEmpty() {
        val customer = Customer(1, "", "")
        Assert.assertFalse(customer.isValid)
    }

    @Test
    fun shouldReturnInvalidCustomer_WhenAllFieldsAreNull() {
        val customer = Customer(1, null, null)
        Assert.assertFalse(customer.isValid)
    }
}